src/main/java/org/springframework/data/repository/reactive/ReactiveSortingRepository.java

```diff
@@ -48,4 +49,35 @@ public interface ReactiveSortingRepository<T, ID> extends Repository<T, ID> {
 	 * @throws IllegalArgumentException in case the given {@link Sort} is {@literal null}.
 	 */
 	Flux<T> findAll(Sort sort);
+
+	/**
+	 * Returns all entities sorted by the given options with limit.
+	 *
+	 * @param sort the {@link Sort} specification to sort the results by, can be {@link Sort#unsorted()}, must not be
+	 *          {@literal null}.
+	 * @param limit the {@link Limit} to limit the results, must not be {@literal null} or negative.
+	 * @return all entities sorted by the given options.
+	 * @throws IllegalArgumentException in case the given {@link Sort} is {@literal null} or {@link Limit} is
+	 *           {@literal null} or negative.
+	 */
+	default Flux<T> findAll(Sort sort, Limit limit) {
+		if (limit == null) {
+			throw new IllegalArgumentException("Limit must not be null");
+		}
+
+		if (limit.isUnlimited() || limit.max() == Integer.MAX_VALUE) {
```

Comment:
```suggestion
        final int maxLimit = limit.max(); // Moved from L72
		if (limit.isUnlimited() || maxLimit == Integer.MAX_VALUE) {
```
How about invoke `limit.max` once?

Reply:
When the`max()` method is called on an `Unlimited` instance, it throws an `IllegalStateException`.

```java
@Override
public int max() {
	throw new IllegalStateException(
			"Unlimited does not define 'max'. Please check 'isLimited' before attempting to read 'max'");
}
```

Would it be better if i separate the two conditions into individual if-statements?

Reply:
oh~ you're right. just ignore my comment 🙇